includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -355,6 +362,9 @@ public function __construct(
 			'link'          => 'link_payments',
 		];
 
+		// Platform checkout utilities.
+		$this->platform_checkout_util = new Platform_Checkout_Utilities();
```

Comment:
This sounds like a good suggestion and I had already made the change just to realize that this will need me to fix 200+ tests. I am planning to do this refactor in a separate PR later to keep this PR decluttered. :grimacing:

Reply:
Ooh, gotcha. That's definitely not worth the effort in this PR. 

> The benefit of this approach is that it would now enable us to mock out all of the Platform_Checkout_Utilities functions if desired. 

Dependency injection is helpful, because it gives us more control over how our dependencies are attached, specifically allowing us to mock them out relatively easily in our tests...which is only really important, if it's beneficial to mock out particular functions performed by the dependency in question. 

In this class, we only rely on the `Platform_Checkout_Utilities::should_save_platform_customer()` function in this dependency. This function basically performs a check based on the contents of the `$_POST` global and the `WC()` session data. It's up to you whether it makes more sense to simply manually fill these variables or mock out the function entirely: if you decide on the former, then there's no real benefit to injecting the dependency (for now) and--considering the exttrenuous effort--it's probably not worth the trouble.

Aaaanyways, as far as I can see right now, there aren't any present tests that test the differing outcomes of this function at all. I'll leave it up to you to decide whether there should be and in turn whether this entire refactor might bear any fruit or whether this whole discussion can be safely forgotten. 😅